packages/backend/src/modules/tracked-txs/TrackedTxsIndexer.ts

```diff
@@ -38,88 +42,98 @@ export class LivenessIndexer extends ChildIndexer {
   }
 
   override async update(from: number, to: number): Promise<number> {
-    const [configurations, adjustedTo] = await this.getConfiguration(from, to)
+    const [configurations, syncTo] = await this.getConfiguration(from, to)
 
     if (configurations.length === 0) {
-      this.logger.debug('Update skipped', { from, to: adjustedTo })
-      return adjustedTo.toNumber()
+      this.logger.debug('Update skipped', { from, to: syncTo })
+      return syncTo.toNumber()
     }
 
-    const data = await this.livenessClient.getLivenessData(
+    const txs = await this.trackedTxsClient.getData(
       configurations,
       new UnixTime(from),
-      adjustedTo,
+      syncTo,
     )
 
-    await this.livenessRepository.runInTransaction(async (trx) => {
-      await this.livenessRepository.addMany(data, trx)
+    await this.configRepository.runInTransaction(async (trx) => {
+      for (const [type, updater] of Object.entries(this.updaters)) {
+        const filteredTxs = txs.filter((tx) => tx.use.type === type)
+        await updater.update(filteredTxs, trx)
+      }
 
-      await this.configurationRepository.setLastSyncedTimestamp(
-        configurations.map((c) => c.id),
-        adjustedTo,
-        trx,
+      const configIds = configurations.flatMap((config) =>
+        config.uses.map((use) => use.id),
       )
+      await this.configRepository.setLastSyncedTimestamp(configIds, syncTo, trx)
     })
 
     this.logger.info('Updated', {
       from,
-      adjustedTo: adjustedTo,
+      adjustedTo: syncTo,
       usedConfigurations: configurations.length,
-      fetchedDataPoints: data.length,
+      fetchedTxsCount: txs.length,
     })
-    return adjustedTo.toNumber()
+    return syncTo.toNumber()
   }
 
   async getConfiguration(
     from: number,
     to: number,
-  ): Promise<[LivenessConfigEntry[], UnixTime]> {
+  ): Promise<[TrackedTxConfigEntry[], UnixTime]> {
     const adjustedTo = adjustToForBigqueryCall(from, to)
```

Comment:
.getConfiguration does too many things, I think we can move this line to .update